evaluation/evaluate.py

```diff
@@ -194,81 +196,224 @@ def run_evaluation_from_config(working_dir: Path, config: dict, num_questions: i
         logger.error("Evaluation was terminated early due to an error ⬆")
 
 
+def plot_bar_charts(
+    layout: Tuple[int, int],
+    data: List[Dict[str, any]],
+    titles: List[str],
+    y_labels: List[str],
+    y_max_lim: List[float] = None,
+    width=0.4,
+    output_path: Path = Path("evaluation_results.png"),
+):
+    """Plot bar charts for the provided data."""
+    assert layout[0] * layout[1] == len(data), "Number of data points must match the layout"
+
+    font = {"weight": "normal", "size": 9}
+
+    matplotlib.rc("font", **font)
+
+    fig, axs = plt.subplots(layout[0], layout[1], figsize=(layout[1] * 5, layout[0] * 4))
+    fig.tight_layout(pad=3.0)
+
+    for i, ax in enumerate(axs.flat):
+        x_data = list(data[i].keys())
+        y_data = list(data[i].values())
+        name = titles[i]
+        colors = plt.cm.tab20.colors[: len(x_data)]
+
+        ax.bar(x_data, y_data, width=width, label=name, color=colors)
+        ax.set_title(name, pad=20)
+        if y_max_lim is not None and len(y_max_lim) > i and y_max_lim[i] is not None:
+            ax.set_ylim(0, max(y_max_lim[i], max(y_data)))
+        else:
+            ax.set_ylim(0, np.ceil(max(y_data)) * 1.2)
+
+        ax.set_ylabel(y_labels[i])
+
+        for j, v in enumerate(y_data):
+            ax.text(j, v * 1.02, str(round(v, 2)), ha="center")
+
+    plt.savefig(output_path)
+    plt.close(fig)
+
+
+def plot_box_charts(
+    layout: Tuple[int, int], data: List[List[float]], titles: List[str], y_labels: List[str], output_path: Path
+):
+    """Plot box charts for the provided data."""
+    assert layout[0] * layout[1] == len(data), "Number of data points must match the layout"
```

Comment:
Same note as above, we don't typically raise `AssertionErrors` outside of tests when we can have much richer errors. I'd probably switch to sth like this:
```suggestion
    if layout[0] * layout[1] != len(data):
        raise ValueError("Number of data points must match the layout")
```

Reply:
same above